datalad/plugin/addurls.py

```diff
@@ -856,6 +862,23 @@ def __call__(dataset, urlfile, urlformat, filenameformat,
                         "ds": ds_current,
                         "ds_filename": ds_filename})
 
+        if version_urls:
+            lgr.info("Versioning URLs")
+            pbar = ui.get_progressbar(total=len(rows),
+                                      label="Versioning URLs", unit=" URLs")
+            for row in rows:
+                try:
+                    row["url"] = get_versioned_url(row["url"])
+                except (ValueError, NotImplementedError) as exc:
+                    # We don't expect this to happen because get_versioned_url
+                    # should return the original URL if it isn't an S3 bucket.
+                    # It only raises exceptions if it doesn't know how to
+                    # handle the scheme for what looks like an S3 bucket.
+                    lgr.warning("error getting version of %s: %s",
+                                row["url"], exc_str(exc))
+                pbar.update(1, increment=True)
```

Comment:
Right, there are already other places in addurls that add put successes/failure in the bar.  I decided not to do that because the current code works with status dicts but I didn't want to actually yield those results (since in your use case there are thousands of URLs and AFAICS failure should be rare).  Of course that doesn't prevent me from either 1) abusing the status dicts for the progress bar but not yielding them or 2) explicitly writing code to handle the `set_desc` in this case; it's just what made me think "meh, not worth it".  Or perhaps I really should be yielding these results?  Thoughts?

Reply:
Just FTR, this could be easily implemented using the log->pbar approach for which there is an open PR